app/controllers/admin/health/patient_referrals_controller.rb

```diff
@@ -168,6 +177,25 @@ def assign_agency
     end
     # rubocop:enable Style/IfInsideElse
 
+    private def assign_agency_inner(patient_referral, agency_id = nil, care_coordinator_id = nil, nurse_care_manager_id = nil)
```

Comment:
They are optional because there are a few cases where this is called:

1) The PR was previously unassigned, and user is assigning it. In this case, the agency is inferred from the CC or the NCM. (Tab 1)
2) The PR was previously assigned, and user is re-assigning it to a different agency. In this case, the `agency_id` is present, and CC/NCM are nil.
3) The PR was previously assigned, and user is un-assigning it. In this case, all three of agency_id/CC/NCM are nil.

I'm not sure how to make this clearer in the code, but I can add it as a comment?

Reply:
OK -- my quick search didn't find any place you called `assign_agency_inner` without all 4 args... Must have missed it... This is really a matter of personal taste, but rather than a bunch of optional positional arguments, I'd prefer:
```private def assign_agency_inner(patient_referral, agency_id: nil, care_coordinator_id: nil, nurse_care_manager_id: nil)```

Reply:
Ok cool I didn't know about keyword arguments like that. Updated